### DISCUSSION TITLE
What is the recommended technique for obtaining rendered videos of episodes?

### DISCUSSION THREAD

**Discussion by vrn-sn** [PKU-Alignment/safety-gymnasium#102] [Q&A]
It looks like `gymnasium.wrappers.Monitor` is no longer supported, and the following `RecordVideo` wrapper doesn't seem to internally support the `cost` output that is returned when `step` is called:

```python3
from gymnasium.wrappers.record_video import RecordVideo

env = safety_gymnasium.make("SafetyRacecarGoal1-v0", render_mode="rgb_array")
env = RecordVideo(env, './videos')
```

I've also seen others render videos from a 3rd-person perspective, but I can only manage to get 1st-person perspective videos rendered. Does anyone know how to change this?

**Comment by muchvo** [comment]
Please refer to documentation for [API](https://www.safety-gymnasium.com/en/latest/introduction/basic_usage.html#render) and [camera options](https://www.safety-gymnasium.com/en/latest/api/bases/underlying.html#safety_gymnasium.bases.underlying.Underlying.render) on rendering the image, and follow this example file for saving [images accordingly](https://github.com/PKU-Alignment/safety-gymnasium/blob/main/examples/vision_env.py). For complete example, you can refer to [here](https://github.com/PKU-Alignment/omnisafe/blob/HEAD/omnisafe/evaluator.py#L442).
